src/caterpillar-game-react.tsx

```diff
@@ -15,24 +15,25 @@ interface DailyClue {
   categories: Category[];
 }
 
+// hour each clue is released
+const clueHours = [7, 11, 15, 19, 23]
+
 // Assert the type of dailyCluesData
 const dailyClues: DailyClue = dailyCluesData as DailyClue;
 
 const CaterpillarGame: React.FC = () => {
+  const currentHour = new Date().getHours()
+  const visibleClues = clueHours.filter(hour => currentHour >= hour).length - 1 // -1 to make it 0-indexed
```

Comment:
`-1` to make `visibleClues` 0-indexed because the comparison below with `clueIndex` which is an array index and also 0-indexed.